### PR TITLE
Refactors

### PR DESCRIPTION
**Pull request by MarcelWilson**

### CODE REVIEW

File: clockify_client/models/project.py [hunk 1]
```diff
@@ -21,17 +21,18 @@ def get_projects(self, workspace_id: str, params: dict | None = None) -> JsonTyp
...
 
-        return self.get(url)
+        return self.get(path)
 
     def add_project(
         self,
         workspace_id: str,
         project_name: str,
         client_id: str,
+        *,
```
Comment (a-rhote) [inline clockify_client/models/project.py:35]: Why are we accepting arbitrary arguments here?

File: clockify_client/models/task.py [hunk 2]
```diff
@@ -16,44 +16,43 @@ def add_task(
...
         """
-        url = f"{self.base_url}/workspaces/{workspace_id}/projects/{project_id}/tasks/"
-        payload = {"name": task_name, "projectId": project_id}
-        if request_data:
-            payload = {**payload, **request_data}
+        path = f"/workspaces/{workspace_id}/projects/{project_id}/tasks/"
 
-        return self.post(url, payload)
+        _payload = {"name": task_name, "projectId": project_id}
+        payload = {**_payload, **payload} if payload else _payload
```
Comment (a-rhote) [inline clockify_client/models/task.py]: Don't dictionaries have an extend method like lists do? You could just `_payload.extend(payload)` here and then pass that in to the post below.

File: clockify_client/models/task.py [hunk 3]
```diff
@@ -66,14 +65,14 @@ def get_tasks(
         :param params        Request URL query parameters.
         :return              List with dictionaries with task object representation.
         """
-        _url = f"{self.base_url}/workspaces/{workspace_id}/projects/{project_id}"
+        _url = f"/workspaces/{workspace_id}/projects/{project_id}"
```
Comment (a-rhote) [inline clockify_client/models/task.py]: I'd rename this base_path rather than _url

File: clockify_client/models/time_entry.py [hunk 4]
```diff
@@ -22,15 +22,15 @@ def get_time_entries(
         :param params       Request URL query params.
         :return  List with dictionary representation of time entries from clockify.
         """
-        _url = f"{self.base_url}/workspaces/{workspace_id}/user/{user_id}"
+        _path = f"/workspaces/{workspace_id}/user/{user_id}/time-entries"
```
Comment (a-rhote) [inline clockify_client/models/time_entry.py]: Same here

File: tests/test_clockify.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+from clockify_client import Clockify
+
+
+def test_can_be_instantiated() -> None:
+    clockify = Clockify("apikey", "baz.co")
+    assert isinstance(clockify, Clockify)
+
+
+def test_singleton() -> None:
```
Comment (a-rhote) [inline tests/test_clockify.py]: We should rename this test, you're not testing it's a singleton here, you're honestly testing its not a singleton.

**Review by a-rhote — CHANGES_REQUESTED** [review]

File: clockify_client/models/project.py (hunk 1 above)
Comment (MarcelWilson) [reply clockify_client/models/project.py:35]: This confused me too the first time I saw it.  It forces callers to use kwargs for all the arugments after the `*`.  It is one way to solve the boolean trap pattern.

Comment (a-rhote) [reply clockify_client/models/project.py:35]: Gooootcha! Okay.

File: clockify_client/models/task.py (hunk 2 above)
Comment (MarcelWilson) [reply clockify_client/models/task.py]: I can use `update` instead of splatting.

File: clockify_client/models/task.py (hunk 3 above)
Comment (MarcelWilson) [reply clockify_client/models/task.py]: Fixed

File: tests/test_clockify.py (hunk 5 above)
Comment (MarcelWilson) [reply tests/test_clockify.py]: Oops.  That's fixed.

**Review by a-rhote — DISMISSED** [review]

**Review by a-rhote — APPROVED** [review]

**Pull request merged**